_posts/en/newsletters/2021-04-14-newsletter.md

```diff
@@ -0,0 +1,225 @@
+
+    3. To use height for the minimum activation parameter.  This further
+       simplifies analysis and also remains compatible with the goal of
+       allowing multiple test networks to share activation parameters.
+       Even though height may differ on those networks, they can all use
+       a minimum activation height of `0` to activate within the window
+       defined by MTP.
+
+    Although some discussion participants expressed their displeasure
+    with the compromise proposal, its [implementation][bitcoin core
+    #21377] has now received reviews or
+    expressions of support from over a dozen active contributors to
+    Bitcoin Core and the maintainers of two other full node
+    implementations (btcd and libbitcoin).  We hope this momentum
+    to activate taproot continues and we'll be able to report additional
+    progress in a future newsletter.
+
+## Bitcoin Core PR Review Club
+
+*In this monthly section, we summarize a recent [Bitcoin Core PR Review Club][]
+meeting, highlighting some of the important questions and answers.  Click on a
+question below to see a summary of the answer from the meeting.*
+
+[Introduce deploymentstatus][review club
+#19438] is a PR ([#19438][Bitcoin Core #19438]) by Anthony Towns that proposes
+three helper functions to make it easier to [bury future deployments without
+changing all the code paths][easier burying] that check [BIP9][] status:
+`DeploymentEnabled` to test if a deployment can be active, `DeploymentActiveAt`
+to check if a deployment should be enforced in the given block, and
+`DeploymentActiveAfter` to know if a deployment should be enforced in the
+following block. All three work with both buried deployments and version bits
+deployments.
+
+The review club discussion focused on understanding the change and its potential
+benefits.
+
+{% include functions/details-list.md
+  q0="What are the advantages of a [BIP90][] buried deployment over a [BIP9][]
+     version bits deployment?"
+  a0="A buried deployment simplifies the deployment logic by replacing the test
+     that governs enforcement with simple height checks, thereby reducing the
+     technical debt associated with deployment of those consensus changes."
+  a0link="https://bitcoincore.reviews/19438#l-132"
+
+  q1="How many buried deployments are defined by this PR?"
+  a1="Five: height in coinbase, CLTV ([BIP65][] `CHECKLOCKTIMEVERIFY`), strict
+     DER signatures, CSV ([BIP112][] `OP_CHECKSEQUENCEVERIFY`), and segwit. They
+     are listed in the `BuriedDeployment` enumerator proposed by the PR in
+     [src/consensus/params.h#L14-22](https://github.com/bitcoin/bitcoin/blob/e72e062e/src/consensus/params.h#L14-L22).
+     One could argue that the [Satoshi-era
+     softforks](/en/topics/soft-fork-activation/#2009-hardcoded-height-consensus-nlocktime-enforcement)
+     are also buried."
+  a1link="https://bitcoincore.reviews/19438#l-75"
+
+  q2="How many version bits deployments are defined by this PR?"
```

Comment:
I'm not sure I'd say that these two version bits deployments are "defined by this PR". They are already defined in the code in master.

Reply:
Thanks @jnewbery! You are right, https://github.com/bitcoin/bitcoin/pull/19438/commits/994b0ce58f832da28adf90ab08ce2a7d6886cdae adds the `BuriedDeployment` enum but `DeploymentPos` already exists. That was an oversight on my part.

The question could be instead "How many version bits deployments are there? Where are they listed in the code?" and in the answer, "listed by the PR" replaced by "enumerated in the codebase"

Reply:
Change looks good to me. Thanks!